uvicorn/main.py

```diff
@@ -506,37 +504,42 @@ def _share_socket(sock: socket) -> socket:
 
         else:
             # Standard case. Create a socket from a host/port pair.
+
+            host = config.host
+            port = config.port
+            addr_format = "%s://%s:%d"
+
+            if host and ":" in host and host[0] == "[":
+                # It's an IPv6 address.
+                host = host.lstrip("[").rstrip("]")  # '[::]' -> '::'.
+                addr_format = "%s://[%s]:%d"
```

Comment:
This condition and stripping of brackets is what we do in HTTPX, see:

https://github.com/encode/httpx/blob/7fda99fcefcd0301fab3e5268919ad463352d4c6/httpx/_models.py#L200-L203

(This can only work if users pass IPv6 addresses in brackets, which is a fair constraint, and is inline with standards _anyway_.)